quickwit-metastore/src/metastore/single_file_metastore.rs

```diff
@@ -735,4 +751,70 @@ mod tests {
             MetastoreError::InternalError { .. }
         ));
     }
+
+    #[tokio::test]
+    async fn test_single_file_metastore_race_condition() {
```

Comment:
Does that test fail (at least sometimes) without your fix?

Reply:
@guilload I changed the test using threads and now it fails sporadically on the old implementation.
Sometimes `SplitDoesNotExit` error, other times published splits don't match the expected.

Reply:
So the tokio tasks was not failing at all. My guess was that tokio is kind of executing these tasks on the same thread and in order to get better randomness, I need at least to find a way to force tokio run it on a different threads or at least submit the task from different threads to increase the chance of random behavior. I went with submitting from different threads.
Thought not so helpful, but I also made a tour on this to learn a bit about the scheduler https://tokio.rs/blog/2019-10-scheduler

Reply:
I see. Thanks for the explanation. Not that you should do it, but it'd be fun to try again with some tweaks: number of threads in the runtime, number of tasks spawned, adding some sleep here and there.

Reply:
@guilload  You were right in asking to try out new things. I found out that tokio during tests uses a `simple-scheduler`. So if we want this kind of behavior, we needed to request it with something like `#[tokio::test(flavor = "multi_thread", worker_threads = 3)]`